Docs: update license restrictions

AgnesToulet opened the pull request:
**What this PR does / why we need it**:
Update license documetation as we are deprecating tiered licensing

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

AgnesToulet committed:
Docs: update license restrictions

gguillotte-grafana commented on docs/sources/enterprise/license/license-restrictions.md:
```diff
@@ -12,135 +12,41 @@ When you become a Grafana Enterprise customer, you receive a license that govern
+- [combined licensing](#combined-licensing), or
+- [tiered licensing](#tiered-licensing-deprecated) (deprecated)
 
-To understand which license type you have, refer to [Determine the number of active users for each licensed role](#determine-the-number-of-active-users-for-each-licensed-role).
```
Why is this line removed? This seems more relevant with the deprecation, so users can determine whether they need to update their licensing.

AgnesToulet replied:
This will no longer be available in 9.0. I guess we could replace this with "contact your Grafana account team"? cc @mjseaman

mjseaman replied:
Agree @AgnesToulet

gguillotte-grafana commented on docs/sources/enterprise/license/license-restrictions.md:
```diff
@@ -12,135 +12,41 @@ When you become a Grafana Enterprise customer, you receive a license that govern
-
-Before you upgrade to Grafana 8.3, ensure that the total number of active users in Grafana does not exceed the number of users in your combined license. If it does, then new users cannot sign in to Grafana 8.3 until the active user count returns below the licensed limit.
-
-## Determining your current license type
```
This section seems more relevant with the deprecation, so users can determine whether they need to update their licensing, and IMO should not be removed.

AgnesToulet replied:
Yes you're right, it's still relevant for users coming from before 8.3 to 9.0, I put it back in the doc.

gguillotte-grafana replied:
I'm sorry, I should have been clearer. The "transition to combined licensing model" section still appears to be relevant. Otherwise, I'm not sure how 9.0 users with tiered licensing will be able to transition to combined licensing.

gguillotte-grafana commented on docs/sources/enterprise/license/license-restrictions.md:
```diff
@@ -12,135 +12,41 @@ When you become a Grafana Enterprise customer, you receive a license that govern
-Users who are synced via SAML, Oauth, or LDAP will also lose these attributes.
+1. Review the utilization count on the **Utilization** panel.
 
-### Update user permissions for dashboards
```
Should there be a note pointing to RBAC, which provides the ability to change the access certain roles of users have to specific dashboards regardless of license type, to replace this functionality being deprecated by tiered licenses?

AgnesToulet replied:
We talked about permissions here before because it had a strong impact on the active user count that was used for  licensing. As it has no impact anymore on licensing, I don't think it's relevant to keep it here anymore.

gguillotte-grafana reviewed (commented):
I'm curious as to why we should remove the docs for determining the license type, since it seems more users should be doing that with the deprecation in order to determine whether they need to update their license.

gguillotte-grafana commented:
This is milestoned to 9.0 and labelled for no backport, but contains content quite relevant to users of 8.3. I suggest backporting to 8.3.

AgnesToulet committed:
apply review feedback

AgnesToulet commented:
This PR doesn't contain any new information for 8.3. I just moved the `Combined licensing` part above the `Tiered licensing` because it should be the most important part of the documentation from 9.0 (as tiered licensing got deprecated).

It's not really easy to read as I removed / moved a lot of things but the only new information I added is:
> As of Grafana Enterprise Version 9.0, tiered licensing will be automatically treated as combined licensing.
For example, your license might include 300 viewers and 50 editors or administrators. This will be handled as a combined license for 350 users.

gguillotte-grafana commented:
I see - thanks, @AgnesToulet. 8.3 users don't need to know that the feature is deprecated in 9.0, that's where I was confused. If users on 9.0 still have tiered licenses and need to manage users until they change to combined licensing, and can't find the information in the 9.0 docs, the docs for 8.3 should still be live.

I'm not sure if users will have the same confusion, but perhaps support should be aware that these docs are leaving in 9.0, but the feature will still be present.

mjseaman committed:
remove initial description of two license types

Clean up tiered licensing copy and keep it all in one deprecated section, refer to old docs if user wants to learn more.

mjseaman reviewed (approved):
Made some copy updates, LGTM! Thanks @AgnesToulet.

AgnesToulet committed:
Merge branch 'main' into docs/licensing-update

AgnesToulet committed:
fix linting issues

AgnesToulet committed:
update title

AgnesToulet commented:
> If users on 9.0 still have tiered licenses and need to manage users until they change to combined licensing, and can't find the information in the 9.0 docs, the docs for 8.3 should still be live.

@gguillotte-grafana This won't happen as, from 9.0, we handle tiered licenses as combined licenses even if the customer didn't upgrade (e.g. if a customer has a tiered licenses with 20 admins and 50 viewers, it will be handled like he has a combined license with 70 users in 9.0).

Also, Mitch did some updates and added this for older versions: 
> If you are running a pre-9.0 version of Grafana Enterprise, please refer to the documentation for that version to learn more about license enforcement in your current version.

AgnesToulet committed:
Merge branch 'main' into docs/licensing-update

AgnesToulet committed:
fix config url
